sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/stringExpressions.scala

```diff
@@ -2538,6 +2538,70 @@ case class Encode(value: Expression, charset: Expression)
     newLeft: Expression, newRight: Expression): Encode = copy(value = newLeft, charset = newRight)
 }
 
+/**
+ * Converts the input expression to a binary value based on the supplied format.
+ */
+// scalastyle:off line.size.limit
+@ExpressionDescription(
+  usage = """
+    _FUNC_(str[, fmt]) - Converts the input `str` to a binary value based on the supplied `fmt`.
+      By default, the binary format for conversion is "hex" if `fmt` is omitted.
+  """,
+  examples = """
+    Examples:
+      > SELECT _FUNC_('abc', 'utf-8');
+       abc
+  """,
+  since = "3.3.0",
+  group = "string_funcs")
+// scalastyle:on line.size.limit
+case class ToBinary(expr: Expression, format: Option[Expression], child: Expression)
+  extends RuntimeReplaceable {
+
+  def this(expr: Expression, format: Expression) = this(expr, Option(format),
+    format match {
+      case lit if lit.foldable => lit.eval()
```

Comment:
We would have to check if it's `null` here too.

Reply:
Thanks, added! Now it behaves as below:

```
scala> spark.sql("select to_binary('abc', null)").show()
+--------------------+
|to_binary(abc, NULL)|
+--------------------+
|                null|
+--------------------+

scala> spark.sql("select to_binary(null, 'hex')").show()
+--------------------+
|to_binary(NULL, hex)|
+--------------------+
|                null|
+--------------------+

scala> spark.sql("select to_binary(null, null)").show()
+---------------------+
|to_binary(NULL, NULL)|
+---------------------+
|                 null|
+---------------------+

```